lib/process-services-cloud/src/lib/process-cloud/services/process-cloud.service.ts

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AlfrescoApiService, FormValues, AppConfigService, LogService } from '@alfresco/adf-core';
+import { Injectable } from '@angular/core';
+import { Observable, from, throwError } from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
+import { ProcessDefinitionRepresentationCloud } from '../models/process-definition-cloud.model';
+import { ProcessInstanceVariableCloud } from '../models/process-instance-variable-cloud.model';
+import { ProcessInstanceCloud } from '../models/process-instance-cloud.model';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ProcessCloudService {
+
+    constructor(private alfrescoApiService: AlfrescoApiService,
+                private appConfigService: AppConfigService,
+                private logService: LogService) {
+    }
+
+    /**
+     * Gets process definitions associated with an app.
+     * @param appId ID of a target app
+     * @returns Array of process definitions
+     */
+    getProcessDefinitions(appName?: string): Observable<ProcessDefinitionRepresentationCloud[]> {
+
+        if (appName) {
+            let queryUrl = this.buildQueryUrl(appName);
+
+            return from(this.alfrescoApiService.getInstance()
+                .oauth2Auth.callCustomApi(queryUrl, 'GET',
+                    null, null, null,
+                    null, null, null, ['application/json'],
+                    ['application/json'], Object, null, null)
+            ).pipe(
+                map(this.extractData),
+                map(processDefs => processDefs.map((pd) => new ProcessDefinitionRepresentationCloud(pd.entry))),
+                catchError(err => this.handleProcessError(err))
+            );
+        } else {
+            this.logService.error('AppName is mandatory for querying task');
+            return throwError('AppName not configured');
+        }
+    }
+
+    /**
+     * Starts a process based on a process definition, name, form values or variables.
+     * @param appName name of the Application
+     * @param processDefinitionId Process definition ID
+     * @param name Process name
+     * @param outcome Process outcome
+     * @param startFormValues Values for the start form
+     * @param variables Array of process instance variables
+     * @returns Details of the process instance just started
+     */
+    startProcess(appName: string,
+                 processDefinitionId: string,
+                 name: string,
+                 outcome?: string,
+                 startFormValues?: FormValues,
+                 variables?: ProcessInstanceVariableCloud[]
+    ): Observable<ProcessInstanceCloud> {
+
+        let requestPayload: any = {
```

Comment:
can you create a model as type?
```
"StartProcessPayload": {
            "type": "object",
            "properties": {
                "businessKey": {
                    "type": "string"
                },
                "id": {
                    "type": "string"
                },
                "processDefinitionId": {
                    "type": "string"
                },
                "processDefinitionKey": {
                    "type": "string"
                },
                "processInstanceName": {
                    "type": "string"
                },
                "variables": {
                    "type": "object"
                }
            },
            "title": "StartProcessPayload"
        },
```

Reply:
"id" field refers to the id of the StartProcessPayload object, not to the Process Instance id.
In any case, the id is generated:
https://github.com/Activiti/activiti-api/blob/04d42fed3d0e2a213168437f103ab7da3a56e174/activiti-api-process-model/src/main/java/org/activiti/api/process/model/payloads/StartProcessPayload.java#L19
So it's gonna be removed from the swagger specification